### DISCUSSION TITLE
GitHub settings template

### DISCUSSION THREAD

**Discussion by cegekaJG** [microsoft/AL-Go#964] [Q&A]
I'd like to ask everybody how you deal with setting up the GitHub repository settings when using the AL Go template.
Since there's a lot of settings that have to be configured for a new repository (such as merging strategies, ruleset, branches, etc), everyone usually asks me to handle the creation whenever it's necessary. I guess the one easy way could be to add a workflow that triggers when the repo is created for the first time. Depending on the organization settings, the repository might be missing the necessary permissions, however.
Another way could be to abandon using templates and use a workflow that creates a new repository instead, similar to the "Deploy"-workflow in this repository.
I don't remember seeing the guide mention anything about inheriting GitHub settings, and I'm sure it's something everyone here is dealing with, so how do you do it?

**Comment by freddydk** [comment]
I am thinking that indirect templates is what will solve this issue (when implemented).
You will create an AL-Go repository and in that one you can add scripts, settings etc - then you can use that repository as the template for another repository.
When you then update your indirect template - the template will be available to the repositories using this indirect template.

**Comment by mazhelez** [comment]
Hey @cegekaJG 
Have you already found a way that works for you regarding your struggle?

As Freddy mentioned, indirect templates (now called custom templates, actually) are a way to centralize your setup.
Another way could be to use `ALGoOrgSettings` org setting that could be shared with the new repository.

**Reply by cegekaJG** [reply]
Thanks for suggesting `ALGoOrgSettings`, I forgot about that option.
For a long time, I've used a fork of the AL-Go repository, in which I customized both actions and the template folders. However, since this became very unwieldy over time, I've opted to use a custom template repository instead.
It works mostly fine, but I do miss a customization that let me define folders and files to sync with the template using `Update AL-Go System Files`. I've made a PR for this feature a while ago, but it was rejected.
I haven't seen something to replace that feature in the release notes or documentation either.
